client/src/tests/Login.test.tsx

```diff
@@ -1,25 +1,58 @@
 import React from "react";
+import { waitFor } from "@testing-library/react";
 import renderer from "react-test-renderer";
 import Enzyme from "enzyme";
 import Adapter from "enzyme-adapter-react-16";
 import Login from "./../pages/Login";
 import { Input } from "@material-ui/core";
+import * as constants from "./../constants/index";
+import { isWeb, microapps } from "./../constants/index";
 
 Enzyme.configure({ adapter: new Adapter() });
 
-it("Login Page renders correctly", () => {
+Object.defineProperty(window, "location", {
+  writable: true,
+  value: { assign: jest.fn() },
+});
+
+it("Login Page renders web UI correctly", () => {
   const tree = renderer.create(<Login />).toJSON();
   expect(tree).toMatchSnapshot();
 });
 
-it("Login Page redirects given valid user", async () => {
+it("Login Page redirects on web given valid user", () => {
   const validUser = "Admin";
   const mountedWrapper = Enzyme.mount(<Login />);
   const input = mountedWrapper.find("#username").last();
-  Object.defineProperty(window, "location", {
-    writable: true,
-    value: { assign: jest.fn() },
-  });
   input.simulate("blur", { target: { value: validUser } });
   expect(window.location.assign).toBeCalledWith("/home");
 });
+
+it("Login Page renders microapps UI correctly", () => {
+  // Set isWeb to true to render web UI
+  const origIsWeb = constants.isWeb;
+  constants.isWeb = false;
+  const tree = renderer.create(<Login />).toJSON();
+  expect(tree).toMatchSnapshot();
+  constants.isWeb = origIsWeb;
+});
+
+it("Login Page calls microapps Identity API to login", async () => {
+  // Interject our definitions for constant variables
+  const origIsWeb = constants.isWeb;
+  constants.isWeb = false;
```

Comment:
rename the constants package.
constants usually refer to something whose values are fixed, you are not supposed to change it even in test cases. 
Rename it to something like "environmentSetting"

Reply:
moved non-constants into `config.ts` file